crates/vm/src/eval.rs

```diff
@@ -460,46 +133,63 @@ pub fn build_query(mut result: Box<IterRows>, query: Vec<Query>) -> Result<Box<I
     Ok(result)
 }
 
-/// Optimize & compile the [Expr] for late execution
+/// Optimize & compile the [CrudExpr] for late execution
 #[tracing::instrument(skip_all)]
-pub fn build_ast<P: ProgramVm>(p: &mut P, ast: Expr) -> Result<Code, ErrorVm> {
-    let ast = optimize(p, ast)?;
-    compile(p, ast)
+pub fn build_ast(ast: CrudExpr) -> Code {
+    compile_query_expr(ast.optimize(&|_, _| i64::MAX))
```

Comment:
We should only ever pass in `&|_, _| i64::MAX` when testing. Otherwise we want to use the table size metrics by default.